File: src/main/java/com/manson/fo76/service/NoopPasswordEncoder.java
package com.manson.fo76.service;

import org.apache.commons.lang3.StringUtils;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@Component
public class NoopPasswordEncoder implements PasswordEncoder {

  @Override
  public String encode(CharSequence rawPassword) {
    return rawPassword.toString();
  }

  @Override
  public boolean matches(CharSequence rawPassword, String encodedPassword) {
    return StringUtils.equals(rawPassword.toString(), encodedPassword);
  }
}

File: src/main/java/com/manson/fo76/service/NoopPasswordEncoder.kt
package com.manson.fo76.service

import org.apache.commons.lang3.StringUtils
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component

@Component
class NoopPasswordEncoder : PasswordEncoder {
    override fun encode(rawPassword: CharSequence): String {
        return rawPassword.toString()
    }

    override fun matches(rawPassword: CharSequence, encodedPassword: String): Boolean {
        return StringUtils.equals(rawPassword.toString(), encodedPassword)
    }
}